probe-rs/src/architecture/arm/core/m33.rs

```diff
@@ -204,8 +204,8 @@ impl<'probe> CoreInterface for M33<'probe> {
     fn set_breakpoint(&mut self, bp_unit_index: usize, addr: u32) -> Result<(), Error> {
         let mut val = FpCompX::from(0);
 
-        // clear bits which cannot be set
-        let comp_val = addr & 0xff_ff_ff_fe;
+        // clear bits which cannot be set and shift into position
+        let comp_val = (addr & 0xff_ff_ff_fe) >> 1;
```

Comment:
Is there a specific reason you shift this this by 1 instead of making it `0x7F_FF_FF_FF`?

Reply:
To align the address in the register it needed to be shifted.  This is due to the bitfield macro which shifts it back up before inserting the value.  Note how the shift is applied to both the addr and the mask.

The masking could also be removed since after the shift the masked bit would be dropped.  I left it in there so it was explicit that the bit was zeroed out as per the reference manual.

Here is the snippet from the [Armv8-M Architecture Reference Manual](https://developer.arm.com/documentation/ddi0553/bn) Section D1.2.107 it states (with some ephasis):

**BPADDR, bits [31:1]** Breakpoint address. Specifies _bits[31:1]_ of the breakpoint instruction address.
**BE, bit [0]** Breakpoint enable.

Reply:
Ok, I am very silly ... sorry for that, thanks a lot =)